src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -1251,4 +1251,1192 @@ static class C
+            //         M(in 7);
+            Diagnostic(ErrorCode.ERR_RefReturnLvalueExpected, "7").WithLocation(7, 14));
+    }
+
+    [Fact]
+    public void RefReadonlyParameter_OutArgument()
+    {
+        var source = """
+            class C
+            {
+                static void M(ref readonly int p) => System.Console.WriteLine(p);
+                static readonly int x = 5;
+                static void Main()
+                {
+                    M(out x);
+                    int y;
+                    M(out y);
+                    M(out 6);
+                }
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics(
+            // (7,15): error CS0199: A static readonly field cannot be used as a ref or out value (except in a static constructor)
+            //         M(out x);
+            Diagnostic(ErrorCode.ERR_RefReadonlyStatic, "x").WithLocation(7, 15),
+            // (9,15): error CS1615: Argument 1 may not be passed with the 'out' keyword
+            //         M(out y);
+            Diagnostic(ErrorCode.ERR_BadArgExtraRef, "y").WithArguments("1", "out").WithLocation(9, 15),
+            // (10,15): error CS1510: A ref or out value must be an assignable variable
+            //         M(out 6);
+            Diagnostic(ErrorCode.ERR_RefLvalueExpected, "6").WithLocation(10, 15));
+    }
+
+    [Fact]
+    public void PassingParameters_In_RefReadonly_PlainArgument()
+    {
+        var source = """
+            class C
+            {
+                static void M1(in int p)
+                {
+                    M2(p);
+                }
+                static void M2(ref readonly int p) => System.Console.Write(p);
+                static void Main() => M1(5);
+            }
+            """;
+        var verifier = CompileAndVerify(source, expectedOutput: "5");
+        verifier.VerifyDiagnostics(
+            // (5,12): warning CS9503: Argument 1 should be passed with 'ref' or 'in' keyword
+            //         M2(p);
+            Diagnostic(ErrorCode.WRN_ArgExpectedRefOrIn, "p").WithArguments("1").WithLocation(5, 12));
+        verifier.VerifyIL("C.M1", """
+            {
+              // Code size        7 (0x7)
+              .maxstack  1
+              IL_0000:  ldarg.0
+              IL_0001:  call       "void C.M2(ref readonly int)"
+              IL_0006:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void PassingParameters_In_RefReadonly_RefArgument()
+    {
+        var source = """
+            class C
+            {
+                static void M1(in int p)
+                {
+                    M2(ref p);
+                }
+                static void M2(ref readonly int p) => System.Console.Write(p);
+                static void Main() => M1(5);
+            }
+            """;
+        // PROTOTYPE: Should not be an error when value checks are relaxed. Verify execution and IL.
```

Comment:
>// PROTOTYPE: Should not be an error when value checks are relaxed. Verify execution and IL. [](http://example.com/codeflow?start=8&length=92)

I think the error is correct. It is not allowed to take a writable reference of an `in` parameter.

Reply:
But the parameter is `ref readonly`, so the reference is not writable. The callsite modifier `ref` is allowed and equivalent to `in` for `ref readonly` modifiers per the proposal.

Reply:
Clarified offline that this indeed should be an error, will fix.